modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -2274,46 +2260,21 @@ open class PlaybackManager @Inject constructor(
     }
 
     private fun verifySleepTimeForEndOfChapter() {
-        val currentChapterUuid = getCurrentChapterUuid()
-        val currentEpisodeUui = getCurrentEpisode()?.uuid
-
-        if (!isSleepAfterChapterEnabled()) {
-            updateLastListenedState { copy(chapterUuid = null, episodeUuid = null) }
-            return
-        }
-
-        if (playbackStateRelay.blockingFirst().lastListenedState.chapterUuid.isNullOrEmpty()) {
-            updateLastListenedState { copy(chapterUuid = currentChapterUuid) }
-        }
-
-        if (playbackStateRelay.blockingFirst().lastListenedState.episodeUuid.isNullOrEmpty()) {
-            updateLastListenedState { copy(episodeUuid = currentEpisodeUui) }
-        }
-
-        // When we switch from a episode that contains chapters to another one that does not have chapters
-        // the current chapter is null, so for this case we would need to verify if the episode changed to update the sleep timer counter for end of chapter
-        if (currentChapterUuid.isNullOrEmpty() && !playbackStateRelay.blockingFirst().lastListenedState.episodeUuid.isNullOrEmpty() && playbackStateRelay.blockingFirst().lastListenedState.episodeUuid != currentEpisodeUui) {
-            applicationScope.launch {
-                updateLastListenedState { copy(episodeUuid = currentEpisodeUui) }
-                sleepEndOfChapter()
-            }
-        } else if (playbackStateRelay.blockingFirst().lastListenedState.chapterUuid == currentChapterUuid) { // Same chapter
-            return
-        } else { // Changed chapter
-            applicationScope.launch {
-                updateLastListenedState { copy(chapterUuid = currentChapterUuid, episodeUuid = getCurrentEpisode()?.uuid) }
-                sleepEndOfChapter()
-            }
-        }
-    }
+        applicationScope.launch {
+            sleepTimer.verifySleepTimeForEndOfChapter(
+                currentChapterUuid = getCurrentChapterUuid(),
+                currentEpisodeUuid = getCurrentEpisode()?.uuid,
+                onSleepEndOfChapter = {
+                    showToast(application.getString(LR.string.player_sleep_time_fired_end_of_chapter))
+
+                    val podcast = playbackStateRelay.blockingFirst().podcast
+                    if (podcast != null && podcast.skipLastSecs > 0) {
```

Comment:
It would be good to include a comment about why this only pauses if the podcast skip last seconds is greater than zero because I'm not quite sure why it does this.

Reply:
I agree and I've added a comment for it. I had to test this again because this code was already there and I was not sure why we needed this